src/Network/Network/AzureFirewallPolicy/FilterRule/NewAzureFirewallPolicyFilterRuleCollectionCommand.cs

```diff
@@ -0,0 +1,70 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Network.Models;
+using MNM = Microsoft.Azure.Management.Network.Models;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "FirewallPolicyFilterRuleCollection", SupportsShouldProcess = true), OutputType(typeof(PSAzureFirewallPolicyRuleCollectionGroup))]
+    public class NewAzureFirewallPolicyFilterRuleCollectionCommand : NetworkBaseCmdlet
+    {
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The name of the Application Rule Collection")]
+        [ValidateNotNullOrEmpty]
+        public virtual string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The priority of the rule collection")]
+        [ValidateRange(100, 65000)]
+        public uint Priority { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The list of application rules")]
+        [ValidateNotNullOrEmpty]
+        public PSAzureFirewallPolicyRule[] Rule { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The action of the rule collection")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(
```

Comment:
I remember PowerShell team asked us to use `ArgumentCompleter` instead of `ValidateSet`. Is it still recommended, @markcowl?

Reply:
We have a restriction on the values the user can enter here. We allow the values "Allow" and "Deny". As the use case here is not about completing/suggesting the parameter name but about validating the values possible, I am keeping it as validateSet and not changing it. 

Please let me know if you still have any comment about this.